### PR TITLE
Expression based cursor

### PR DESCRIPTION
**Pull request by richardrissanen**
Found this solution via the repos issues here: https://github.com/duffelhq/paginator/issues/173

### CODE REVIEW

**Review by pathaknv — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -140,6 +140,51 @@ cursor_before = metadata.before
...
+      ]
+    )
+    query
+    |> Repo.paginate(
+      limit: 30,
+      fetch_cursor_value_fun: fn
+        # Here we build the rank_value for each returned row
+        schema, :rank_value ->
+          {:ok, %{rows: [[rank_value]]}} =
+            Repo.query("SELECT ts_rank($1, plainto_tsquery('simple', $2))", [
```
Comment (jonbirke0927) [inline README.md:167]: @richardrissanen does this execute a new query per row?

Comment (richardrissanen) [reply README.md:167]: That example does but I am sticking with our function based solution. However, there is a select merge that is required with this solution.

Comment (richardrissanen) [reply README.md:167]: @jonbirke0927 ^^^

Comment (jonbirke0927) [reply README.md:167]: @richardrissanen got it. I think given the fact that this is a PR against the existing repo, and seems others have used it w success, I'm much less worried about the change

**Review by jonbirke0927 — APPROVED** [review]

**Pull request merged**